exporters/trace/src/main/java/com/google/cloud/opentelemetry/trace/InternalTraceExporter.java

```diff
@@ -0,0 +1,174 @@
+/*
+ * Copyright 2022 Google
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.cloud.opentelemetry.trace;
+
+import static com.google.api.client.util.Preconditions.checkNotNull;
+
+import com.google.api.gax.core.FixedCredentialsProvider;
+import com.google.api.gax.core.NoCredentialsProvider;
+import com.google.api.gax.grpc.GrpcTransportChannel;
+import com.google.api.gax.rpc.FixedTransportChannelProvider;
+import com.google.api.gax.rpc.HeaderProvider;
+import com.google.auth.Credentials;
+import com.google.auth.oauth2.GoogleCredentials;
+import com.google.cloud.trace.v2.TraceServiceClient;
+import com.google.cloud.trace.v2.TraceServiceSettings;
+import com.google.cloud.trace.v2.stub.TraceServiceStub;
+import com.google.common.collect.ImmutableMap;
+import com.google.devtools.cloudtrace.v2.AttributeValue;
+import com.google.devtools.cloudtrace.v2.ProjectName;
+import com.google.devtools.cloudtrace.v2.Span;
+import io.grpc.ManagedChannelBuilder;
+import io.opentelemetry.sdk.common.CompletableResultCode;
+import io.opentelemetry.sdk.trace.data.SpanData;
+import io.opentelemetry.sdk.trace.export.SpanExporter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import javax.annotation.Nonnull;
+
+class InternalTraceExporter implements SpanExporter {
+
+  private final CloudTraceClient cloudTraceClient;
+  private final ProjectName projectName;
+  private final String projectId;
+  private final TraceTranslator translator;
+
+  private static final Map<String, String> HEADERS =
+      Map.of("User-Agent", "opentelemetry-operations-java/" + TraceVersions.EXPORTER_VERSION);
+  private static final HeaderProvider HEADER_PROVIDER = () -> HEADERS;
+
+  private static InternalTraceExporter createWithClient(
+      String projectId,
+      CloudTraceClient cloudTraceClient,
+      ImmutableMap<String, String> attributeMappings,
+      Map<String, AttributeValue> fixedAttributes) {
+    return new InternalTraceExporter(
+        projectId, cloudTraceClient, attributeMappings, fixedAttributes);
+  }
+
+  static SpanExporter createWithConfiguration(TraceConfiguration configuration) throws IOException {
+    String projectId = configuration.getProjectId();
+    TraceServiceStub stub = configuration.getTraceServiceStub();
+
+    // TODO: Remove stub.
```

Comment:
As mentioned this is the current existing `TraceExporter` class from the main. This TODO has been there for some while now, but I have not looked into why this is needed. But I agree that it should be handled in a separate bug, since it warrants some investigation.

Reply:
Created issue #198 to track this and linked the issue number in the code.